challenges/functions.js

```diff
@@ -6,32 +6,43 @@
   * The last parameter accepts a callback 
   * In the body of the function return the callback with the two parameters that you created
 */
-function consume(param1, param2, callback){
-
- callback(param1, prama2);
+function consume (param1, param2, callback){
+  return callback(param1, param2);
 }
 
-
-
+ 
 /* Step 2: Create several functions to callback with consume();
   * Create a function named add that returns the sum of two numbers
   * Create a function named multiply that returns the product of two numbers 
   * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
 */
-function add(num1, pa, consume) {
- return consume(num1 + param2);
-}  
+function add(param1, param2) {
+  return param1 + param2;
+ 
+}
+ 
+function multiply(param1, param2) {
+  return param1 * param2;
+ 
+}
+
+function greeting(param1, param2) {
+  return `Hello ${param1}  ${param2}, nice to meet you!`;
+ 
+}
 
 /* Step 3: Check your work by un-commenting the following calls to consume(): */
 consume(2,2,add); // 4
+console.log(consume(2, 2, add));
 consume(10,16,multiply); // 160
+console.log(consume(10, 16, multiply));
 consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
-
+console.log(consume("Mary","Poppins", greeting));
 
 // ==== Closures ==== 
 
 // Explain in your own words why `nestedfunction()` can access the variable `internal`.
-
+// - It can only go up one level and it has to be in the same block of code
 // Explanation: 
```

Comment:
it can go up multiple levels, note that in the guided project when we had the third nested object in the kingdom (the suburb i think we called it) it had access to the variable we declared in the kingdoms function block